webapp/components/profile_picture.jsx

```diff
@@ -25,21 +28,96 @@ export default class ProfilePicture extends React.Component {
     }
 
     render() {
+        let email = '';
+        let profileImg = '';
+        let toRender;
         let statusClass = '';
         if (this.props.status) {
             statusClass = 'status-' + this.props.status;
         }
-
-        return (
-            <span className={`status-wrapper ${statusClass}`}>
+        if (this.props.user) {
+            email = this.props.user.email;
+            profileImg = Client.getUsersRoute() + '/' + this.props.user.id + '/image?time=' + this.props.user.update_at;
+            var dataContent = [];
+            dataContent.push(
                 <img
-                    className='more-modal__image'
-                    width={this.props.width}
-                    height={this.props.width}
-                    src={this.props.src}
+                    className='user-popover__image'
+                    src={profileImg}
+                    height='128'
+                    width='128'
+                    key='user-popover-image'
                 />
-            </span>
-        );
+                );
+            const fullname = Utils.getFullName(this.props.user);
+            if (fullname) {
+                dataContent.push(
+                    <div
+                        data-toggle='tooltip'
+                        title={fullname}
+                        key='user-popover-fullname'
+                    >
+                        <p
+                            className='text-nowrap'
+                        >
+                            {fullname}
+                        </p>
+                    </div>
+                        );
+            }
+            if (global.window.mm_config.ShowEmailAddress === 'true' || UserStore.isSystemAdminForCurrentUser() || this.props.user === UserStore.getCurrentUser()) {
```

Comment:
`this.props.user === UserStore.getCurrentUser()` won't always evaluate correctly. Can you use `this.props.user.id === UserStore.getCurrentId()` instead?